pkg/cli/image/extract/extract.go

```diff
@@ -483,6 +483,10 @@ func (o *Options) Run() error {
 }
 
 func layerByEntry(r io.Reader, options *archive.TarOptions, layerInfo LayerInfo, fn TarEntryFunc, allLayers bool, alreadySeen map[string]struct{}) (bool, error) {
+	// Prevents race condition present in vendored version of docker
+	// https://github.com/moby/moby/issues/39859
+	os.Setenv("MOBY_DISABLE_PIGZ", "true")
```

Comment:
This may introduce *new* race conditions; changing the environment in a multi-threaded process can create undefined behavior; see:

https://github.com/rust-lang/rust/pull/24741

Reply:
The situation is certainly better with this fix that it was before this, the panics were unbearable. Would a mutex around this be sufficient to mitigate the risk or is that not sufficient?

Reply:
From a quick look, golang has [a mutex around os.{G,S}etenv](https://github.com/golang/go/blob/a38a917aee626a9b9d5ce2b93964f586bf759ea0/src/syscall/env_unix.go#L18) - so if there's no C threads running, it's generally OK.

But, it'd be safer to hoist this to early in `main()`.